### ISSUE TITLE
Lock not released after  non empty work space fails

### ISSUE THREAD

**Issue by marczis** [hashicorp/terraform#22941]
### Terraform Version
```
Terraform v0.12.8
+ provider.aws v2.29.0
+ provider.null v2.1.2
```

### Terraform Configuration Files
```hcl
AWS - S3 and DynamoDB for locking
```

### Debug Output
Will provide on request

### Crash Output
Standard lock is on

### Expected Behavior
Shall not be lock in place after the non empty work space deletion fails

### Actual Behavior
I tried to delete a non empty workspace, I got a nice warning that it is not empty and I will have resources here and there, if I delete the workspace, but I can do it with -force - which is what I want, but next call tells me there is a lock active. (I can't remove it with terraform error message: 
```
Failed to unlock state: failed to retrieve lock info: unexpected end of JSON input
```
### Steps to Reproduce
Create a workspace,
Do apply something into it, anything will do.
Create a new workspace
Try to delete the first workspace
Try to force delete the first workspace

**Comment by mildwonkey** [comment]
Hi! Thank you for reporting this issue, and I am sorry that you've experienced it.

It will indeed help us to see the debug output, if possible from both commands: when you've force-deleted a workspace and the following command that reports the message you pasted.

Thanks again!

**Comment by marczis** [comment]
Hi Kristin,
I'm afraid I can't provide the logs from the original system. I'm working
in a heavily regulated industry.
The best I can offer to reproduce the issue in a separated environment and
produce the necessary logs, unfortunately this will take some time, as I'm
on a busy travel schedule.
Please keep the ticket open and I try to generate the logs asap, ~by the
end of the week.
Br,
 Peter

**Comment by pselle** [comment]
Fixed by #24085 and will go out in the next 0.12 release.
